validation/src/test/scala/hmda/validation/rules/lar/validity/V545Spec.scala

```diff
@@ -0,0 +1,36 @@
+package hmda.validation.rules.lar.validity
+
+import hmda.model.fi.lar.LoanApplicationRegister
+import hmda.validation.rules.EditCheck
+import hmda.validation.rules.lar.LarEditCheckSpec
+import org.scalacheck.Gen
+
+class V545Spec extends LarEditCheckSpec with BadValueUtils {
+
+  property("Valid when HOEPA status is 2") {
+    forAll(larGen) { lar =>
+      whenever(lar.hoepaStatus == 2) {
+        lar.mustPass
+      }
+    }
+  }
+
+  property("Valid when Lien status != 3") {
+    forAll(larGen) { lar =>
+      whenever(lar.lienStatus != 3) {
+        lar.mustPass
+      }
+    }
+  }
+
+  val invalidHoepaGen: Gen[Int] = Gen.choose(Int.MinValue, Int.MaxValue).filter(_ != 2)
```

Comment:
@kgudel Ignore everything we just said. Your version works well. `forAll` only runs the test when the generator succeeds. So inside the test, you never have to worry about a `None`.